### PR TITLE
nicer UX for users signing in

### PR DESCRIPTION
**Pull request by pedropregueiro**
Make the message to be signed user-friendly.

@werme left URI open as I wasn't sure how best to fetch that (locally and whatnot). Changing it won't affect the backend, so feel free to do what you want with it.

### CODE REVIEW

File: packages/desktop/src/utils/ethereum.js [hunk 1]
```diff
@@ -32,14 +32,20 @@ export const getUserAccounts = async (provider) => {
 };
 
 export const signAddress = async (provider, address) => {
-  const message = {
-    address,
-    signed_at: new Date().toISOString(),
-  };
+  const signedAt = new Date().toISOString();
+  const nonce = Math.floor(Math.random() * (999999 - 100000) + 100000);
```
Comment (pedropregueiro) [inline packages/desktop/src/utils/ethereum.js:36]: if you know a more cryptographic way to do this, go ahead! server just makes sure nonce is in the signed message.

File: packages/desktop/src/utils/ethereum.js [hunk 2]
```diff
@@ -32,14 +32,20 @@ export const getUserAccounts = async (provider) => {
...
+
+  const message = `NewShades wants you to sign in with your web3 account
+${address}
+
+URI: ...
```
Comment (pedropregueiro) [inline packages/desktop/src/utils/ethereum.js:41]: TBD (check PR description)

File: packages/desktop/src/utils/ethereum.js (hunk 1 above)
Comment (werme) [reply packages/desktop/src/utils/ethereum.js:36]: Cool! I’ll replace with [`crypto.getRandomValues`](https://developer.mozilla.org/en-US/docs/Web/API/Crypto/getRandomValues).

**Comment by werme** [comment]
> Make the message to be signed user-friendly.
> 
> @werme left URI open as I wasn't sure how best to fetch that (locally and whatnot). Changing it won't affect the backend, so feel free to do what you want with it.

Awesome! I’ll add the URI thing, then you can merge whenever together with the backend.

**Pull request merged**